src/Verifier/SAW/Cryptol.hs

```diff
@@ -539,115 +543,136 @@ proveProp sc env prop =
+   , ("<<<",          (scGlobalDef, "Cryptol.ecRotL"))        -- {n, ix, a} (fin n, Integral ix) => [n]a -> ix -> [n]a
+   , (">>>",          (scGlobalDef, "Cryptol.ecRotR"))        -- {n, ix, a} (fin n, Integral ix) => [n]a -> ix -> [n]a
+
+     -- -- Sequences primitives
+   , ("#",            (scGlobalDef, "Cryptol.ecCat"))         -- {a,b,d} (fin a) => [a] d -> [b] d -> [a + b] d
+   , ("splitAt",      (scGlobalDef, "Cryptol.ecSplitAt"))     -- {a,b,c} (fin a) => [a+b] c -> ([a]c,[b]c)
+   , ("join",         (scGlobalDef, "Cryptol.ecJoin"))        -- {a,b,c} (fin b) => [a][b]c -> [a * b]c
+   , ("split",        (scGlobalDef, "Cryptol.ecSplit"))       -- {a,b,c} (fin b) => [a * b] c -> [a][b] c
+   , ("reverse",      (scGlobalDef, "Cryptol.ecReverse"))     -- {a,b} (fin a) => [a] b -> [a] b
+   , ("transpose",    (scGlobalDef, "Cryptol.ecTranspose"))   -- {a,b,c} [a][b]c -> [b][a]c
+   , ("@",            (scGlobalDef, "Cryptol.ecAt"))          -- {n, a, ix} (Integral ix) => [n]a -> ix -> a
+   , ("!",            (scGlobalDef, "Cryptol.ecAtBack"))      -- {n, a, ix} (fin n, Integral ix) => [n]a -> ix -> a
+   , ("update",       (scGlobalDef, "Cryptol.ecUpdate"))      -- {n, a, ix} (Integral ix) => [n]a -> ix -> a -> [n]a
+   , ("updateEnd",    (scGlobalDef, "Cryptol.ecUpdateEnd"))   -- {n, a, ix} (fin n, Integral ix) => [n]a -> ix -> a -> [n]a
+
+     -- -- Enumerations
+   , ("fromTo",       (scGlobalDef, "Cryptol.ecFromTo"))
+     --                            -- fromTo : {first, last, bits, a}
+     --                            --           ( fin last, fin bits, last >== first,
+     --                            --             Literal first a, Literal last a)
+     --                            --        => [1 + (last - first)]a
+   , ("fromThenTo",   (scGlobalDef, "Cryptol.ecFromThenTo"))
+     --                            -- fromThenTo : {first, next, last, a, len}
+     --                            --              ( fin first, fin next, fin last
+     --                            --              , Literal first a, Literal next a, Literal last a
+     --                            --              , first != next
+     --                            --              , lengthFromThenTo first next last == len) => [len]a
+
+   , ("error",        (scGlobalDef, "Cryptol.ecError"))       -- {at,len} (fin len) => [len][8] -> at -- Run-time error
+   , ("random",       (scGlobalDef, "Cryptol.ecRandom"))      -- {a} => [32] -> a -- Random values
+   , ("trace",        (scGlobalDef, "Cryptol.ecTrace"))       -- {n,a,b} [n][8] -> a -> b -> b
+   , ("Array::arrayConstant",(scGlobalDef, "Cryptol.ecArrayConstant")) -- {a,b} b -> Array a b
+   , ("Array::arrayLookup",  (scGlobalDef, "Cryptol.ecArrayLookup")) -- {a,b} Array a b -> a -> b
+   , ("Array::arrayUpdate",  (scGlobalDef, "Cryptol.ecArrayUpdate")) -- {a,b} Array a b -> a -> b -> Array a b
+   ])
```

Comment:
This is incorrect.  The `Array` primitives are in their own module, so we should make a separate map for them.  Their names are of the form `PrimIdent arrayName x`.  In general, there shouldn't be any `::` in the names of primitives.

By the way, I just noticed that we don't seem to have a convenience short-cut for the `Array` module (i.e., `arrayPrim = PrimIdent arrayName`), so feel free to add it to `Crytpol.Util.Ident`.  I am fine if you just add it without a pull request, but whatever works for you.